src/lib/index.js

```diff
@@ -495,15 +495,15 @@ lib.isIE = function() {
  * filled array of the given size.
  *
  * @example
- * lib.toObject('nested.test[2].path', 'value');
+ * lib.objectFromPath('nested.test[2].path', 'value');
  * // returns { nested: { test: [null, null, { path: 'value' }]}
  *
  * @param   {string}    path to nested value
  * @param   {*}         any value to be set
  *
  * @return {Object} the constructed object with a full nested path
  */
-lib.toObject = function(path, value) {
+lib.objectFromPath = function(path, value) {
```

Comment:
an even more verbose _plotly-esque_ name would be `Lib.objectFromAttributeString` or `Lib.objectFromAstr`.

@mdtusz I'll leave it up to you to decide what's best.